src/Rules/Deprecations/OverrideDeprecatedPropertyRule.php

```diff
@@ -0,0 +1,64 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Deprecations;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Property;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use function sprintf;
+
+/**
+ * @implements Rule<Property>
+ */
+class OverrideDeprecatedPropertyRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return Property::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (DeprecatedScopeHelper::isScopeDeprecated($scope)) {
+			return [];
+		}
+
+		if (!$scope->isInClass()) {
+			return [];
+		}
+
+		if ($node->isPrivate()) {
+			return [];
+		}
+
+		$class = $scope->getClassReflection();
+
+		$parents = $class->getParents();
+
+		$propertyName = (string) $node->props[0]->name;
+
+		foreach ($parents as $parent) {
+			if (!$parent->hasProperty($propertyName)) {
+				continue;
+			}
+
+			$parentProperty = $parent->getProperty($propertyName, $scope);
+
+			if (!$parentProperty->isDeprecated()->yes()) {
+				return [];
+			}
+
+			return [sprintf(
```

Comment:
Please use the more modern approach with RuleErrorBuilder instead of just returning a string.